components/ai_chat/ai_chat_tab_helper.h

```diff
@@ -77,6 +77,7 @@ class AIChatTabHelper : public content::WebContentsObserver,
 
   // TODO(nullhook): Abstract the data model
   std::vector<ai_chat::mojom::ConversationTurn> chat_history_;
+  std::string article_text_;
```

Comment:
because we need this to build the prompt on each api call. Should I be saving it some other way?

Reply:
If we're no longer attaching the summarized text, then subsequent API calls will include the entire article text, which increases the message length and could impact on cost. What is the benefit of this approach and how does it affect the answers from Claude on subsequent runs?

Reply:
I was under the impression that we were always adding the article text (and the summarized text, which we could remove as redundant). We add the initial prompt to the conversation, right? Also, if we were not, the user would be asking follow up questions on the summarized text, which has been stripped of most details. 

PS: just checked on the ai_chat logs for latest master and we do include the article on each call as the start of the conversation history. Am I missing something?

Reply:
I agree an interesting optimisation would be to produce a long summary and base questions on that, but it would have to be a pretty comprehensive summary, which we might not want to show to the user for UX reasons. But I think we can look at optimisations in a future PR.

Reply:
Okay, so now I understand where the change is. Previously, we used to store the article text along with the prompt in the conversation history itself. However, with the current approach, we no longer store it in the history directly. Instead, we keep the article text separate and attach it along with the prompt on each API call.

The separation of article text in-memory feels like it's adding complexity, requiring additional handling to ensure association with the conversation history on each API call. However, I see a benefit of this is on subsequent API calls. Perhaps a feature: "User-controlled summarization level" where users can switch between summary lengths or toggle between the summarized version and the full article text based on their preferences.

I'm curious if @petemill has thoughts on this separation. If not, then I'll go ahead and approve this PR.

Reply:
I almost forgot to mention this: make sure to clear the article text in the CleanUp function. You will also notice other members being cleared there.

Reply:
done 👍